app/ts/McdaApp/Workspaces/workspacesUtil/workspacesUtil.ts

```diff
@@ -0,0 +1,82 @@
+import IWorkspaceSummary from '@shared/interface/Workspace/IWorkspaceSummary';
+import _ from 'lodash';
+
+export function getLink(workspace: IWorkspaceSummary): string {
+  return (
+    '/workspaces/' +
+    workspace.id +
+    '/problems/' +
+    workspace.defaultSubProblemId +
+    '/scenarios/' +
+    workspace.defaultScenarioId +
+    '/overview'
+  );
+}
+
+export function extractUniqueCriteria(
+  workspaces: IWorkspaceSummary[]
+): string[] {
+  return _.uniq(_.flatMap(workspaces, 'criteria'));
+}
+
+export function extractUniqueAlternatives(
+  workspaces: IWorkspaceSummary[]
+): string[] {
+  return _.uniq(_.flatMap(workspaces, 'alternatives'));
+}
+
+export function filterWorkspaces(
+  workspaces: IWorkspaceSummary[],
+  alternatives: string[],
+  criteria: string[]
+): IWorkspaceSummary[] {
+  const criteriaInLowerCase = _.map(criteria, _.method('toLowerCase'));
```

Comment:
not that it matters probably but one of the primary optimisations would probably be to make these maps instead of lists so lookup for the `includes` checks is faster